tasks/main.yml

```diff
@@ -1,32 +1,55 @@
 ---
-- name: Ensure /root/.aws/ is there
+- name: Update user directory
+  set_fact:
+    aws_cwa_user_path: "/home/{{ aws_cwa_local_user }}"
+  when:
+    - aws_cwa_local_user != "root"
+    - aws_cwa_local_user == "/" # When not default, assume user has overridden.
+
+- name: Ensure {{ aws_cwa_user_path }}/.aws/ is there
   file:
-    path: '/root/.aws/'
+    path: "{{ aws_cwa_user_path }}/.aws/"
     state: 'directory'
     mode: '0700'
-    owner: 'root'
-    group: 'root'
+    owner: "{{ aws_cwa_local_user }}"
+    group: "{{ aws_cwa_local_user }}"
+  when: aws_cwa_on_prem
+
+- name: Remove {{ aws_cwa_user_path }}.aws/config
+  file:
+    path: "{{ aws_cwa_user_path }}/.aws/config"
+    state: 'absent'
+  notify: restart amazon-cloudwatch-agent
+  when: not aws_cwa_on_prem
 
-- name: Add AWS CWA profile to /root/.aws/config
+- name: Remove {{ aws_cwa_user_path }}.aws/credentials
+  file:
+    path: "{{ aws_cwa_user_path }}/.aws/credentials"
+    state: 'absent'
+  notify: restart amazon-cloudwatch-agent
+  when: not aws_cwa_on_prem
+
+- name: Add AWS CWA profile to {{ aws_cwa_user_path }}/.aws/config
   blockinfile:
-    path: '/root/.aws/config'
+    path: "{{ aws_cwa_user_path }}/.aws/config"
     create: 'yes'
-    owner: 'root'
-    group: 'root'
+    owner: "{{ aws_cwa_local_user }}"
+    group: "{{ aws_cwa_local_user }}"
     mode: 0600
     marker: "# {mark} ansible aws-cwa cfg"
     block: |
       [profile AmazonCloudWatchAgent]
       output = json
       region = {{aws_cwa_region|default('us-east-1')}}
   notify: restart amazon-cloudwatch-agent
+  when: aws_cwa_on_prem
```

Comment:
I do not think this needs to be an explicit var to setup for 'on-prem' (needs renaming if we keep) can just check for `aws_cwa_key_access` and `aws_cwa_key_secret` being set

Reply:
I think that's going to hide a bunch of issues, especially as existing inventories ALWAYS have those values defined. Especially with the change, you're asking for on the other PR.

At best they'll be undefined or empty strings, but more likely they'll be set to AWS Ireland deploy access key and secret when setting up Pulsant due to the global default and vault files.

Reply:
We could do;
`aws_cwa_on_prem: "{{ aws_cwa_key_access is not defined or aws_cwa_key_access | length < 0 }}"`

Reply:
That is ok, would clear it up. I do not think  `aws_cwa_key_access` is always defined if you change as I suggest:

```
    aws_cwa_key_access: "{{ host_aws_access_key_id | default(omit) }}"
    aws_cwa_key_secret: "{{ host_aws_secret_access_key | default(omit) }}"
```

`host_aws_access_key_id` is currently only defined in the `aws_scratch` and `feature_dev` inventories, which can probably be removed as these changes are rolled out as those use cases will use the instance role